docs/userguide/campaigns.md

```diff
@@ -1,25 +1,146 @@
+serves as the primary interface for interacting with BayBE as a user. This class is
+responsible for handling the data, making recommendations, adding measurements and
+most other tasks.
 
-Campaigns define and record an experimentation process, i.e. the execution of a
-series of measurements and the iterative sequence of events involved.
+## Creating a campaign
 
-## Attributes and properties
+### Using the campaign class
 
-To construct a new campaign, it is necessary to provide two objects.
-1. **A search space:** The parameter space in which the campaign operates. It can be a purely discrete, purely continuous or hybir space.
-We refer to the [`Searchspace`](baybe.searchspace.core.SearchSpace) class resp. the corresponding [user guide](./searchspace) for more details.
-2. **An objective:** The optimization objective. It is possible to either optimize a single target or to combine different targets. We refer to the [`Objective`](baybe.objective.Objective) class resp. the corresponding [user guide](./objective) for more details.
+When constructing a campaign, it is necessary to provide two objects:
+A search space (see [class](baybe.searchspace.core.SearchSpace) resp. [user guide](./searchspace))
+and an optimization objective (see [class](baybe.objective.Objective) resp. [user guide](./objective)).
+Optionally, it is possible to provide a different strategy
+(see [class](baybe.strategies.base.Strategy) resp. [user guide](./strategy)) and specify
+other aspects of the campaign (see [here](#AM) for details on
+`numerical_measurements_must_be_within_tolerance`).
 
-The following additional aspects are available to further specify a campaign:
-* **A strategy:**: The strategy that is used during the campaign. As a default, a [`TwoPhaseStrategy`](baybe.strategies.composite.TwoPhaseStrategy) is employed. For more details on strategies, see [here](./strategy).
-* **Conducted experiments**: A pandas ``DataFrame`` containing the experimental representation of previously conducted experiments. Preferably, the entries of this ``DataFrame`` were created by a previous campaign using [the campaigns recommend function](baybe.campaign.Campaign.recommend). If no such ``DataFrame`` is provided, it is assumed that no experiments were conducted previously.
-* **Numerical tolerance**: This is a flag for forcing numerical measurements to be within a pre-defined tolerance. Note that the setting of the tolerances is controlled as a part of the respective parameter.
-* **Previously done batches and fits** In case that a campaign builds upon previously condiucted experiments, it is possible to provide the number of previously done batches and fits.
 
-## How to use campaigns
+~~~python
+from baybe import Campaign
 
-We have a detailed example explaining how to use the ``Campaign`` object in our examples, so please see [here](./../../examples/Basics/campaign) for all necessary information.
\ No newline at end of file
+campaign = Campaign(
```

Comment:
Now this is really a bit too generic for my taste :D Since it really just repeats the constructor signature. However, I think I have a solution that offers some nice middleground: in one the next PRs, I want to introduce a few more "convencience constructors" that require less code and imports for simple cases. Once available, we could use them here to keep the code crisp. 

The example could then look as follows:
```python
from baybe import Campaign
from baybe.parameters import NumericalDiscreteParameter
from baybe.targets import NumericalTarget

campaign = Campaign(
    objective=NumericalTarget(name="y", mode="MAX").to_objective(),
    searchspace=NumericalDiscreteParameter(name="x", values=[1, 2, 3]).to_searchspace(),
)
```

What do you think, @Scienfitz @AVHopp

Reply:
Just to be clear: Do you already want changes here resp. how would you want them to look?

As far as I remember, we explicitly discussed this point, and the fact that the user guide will basically just repeat the constructor (which was a point that I also explicitly made!) was fine for everyone.
Also, I know that @Scienfitz explicitly requested to have a single example of the constructor call that also lists the optional arguments.

Reply:
@AdrianSosic Ive seen that you seem to want to place these convenience constructors in may places. While I have nothing against having them, there should be places where the `canonical` way of using an object is described. For instance here and in the README I would not use the convenience constructor, it should rather be mentioned in the user guide of the respective object. Also given that theyre not included why are we discussing them here?

It is also not true that it repeats the constructor signature, because we have other signature contents due to attrs. This should show the 'intended' signature including optionals including potential short comments, imo that is a good design pattern, I dont care so much if its a complete use case, we have examples for that, it should rather be complete about the object

Reply:
Ok, if both of you prefer the generic version, then I will not block it, so feel free to simply resolve. But let me still emphasize: once the campaign attributes are cleaned, this WILL be an exact repetition of the constructor signature, that's why I'm saying, it will bring zero additional information compared to what you'll see when looking at the campaign class page.

Reply:
Fully agree with this, we are on the same page here! And at that point, we will then also re-formulate this.

Reply:
Closing for now. Will revisit in the upcoming PR where we fix the campaign interface